vgen/src/main/java/io/gapi/vgen/go/GoGapicContext.java

```diff
@@ -401,6 +414,40 @@ private GoImport createMessageImport(MessageType messageType) {
   }
 
   /**
+   * Same as getImports, but scoped to the client_test.go file.
+   */
+  public Iterable<String> getTestImports(Interface service) {
+    TreeSet<GoImport> imports = new TreeSet<>();
+
+    // Add non-service-specific imports.
+    imports.add(GoImport.create("golang.org/x/net/context"));
+    imports.add(GoImport.create(GAX_PACKAGE_BASE, "gax"));
+
+    // Add method request-type imports
+    for (Method method : service.getMethods()) {
+      MessageType inputMessage = method.getInputMessage();
+      MessageType outputMessage = method.getOutputMessage();
+      MethodConfig methodConfig =
+          getApiConfig().getInterfaceConfig(service).getMethodConfig(method);
+      imports.add(createMessageImport(inputMessage));
+      if (!isEmpty(method.getOutputType())) {
+        imports.add(createMessageImport(outputMessage));
+      }
+      if (methodConfig.isPageStreaming()) {
+        TypeRef resourceType = methodConfig.getPageStreaming().getResourcesField().getType();
+        if (resourceType.isMessage()) {
+          imports.add(createMessageImport(resourceType.getMessageType()));
+        }
+      }
+    }
+    List<String> result = new ArrayList<>();
+    for (GoImport goImport : imports) {
+      result.add("    " + goImport.importString());
+    }
+    return result;
+  }
```

Comment:
Seems to be a lot of duplication with `getImports`. Do you think it's worth refactoring so we only need to write the `// Add method request-type imports` section only once?